src/axom/sina/include/AdiakWriter.hpp

```diff
@@ -0,0 +1,50 @@
+#ifndef SINA_ADIAK_HPP
+#define SINA_ADIAK_HPP
+
+/// @file
+
+#include "axom/sina/config.hpp"
+#ifdef AXOM_SINA_USE_ADIAK
+
+#include <string>
+#include <type_traits>
+
+#include "axom/sina/include/ConduitUtil.hpp"
+#include "axom/sina/include/Record.hpp"
+#include "axom/sina/include/Run.hpp"
+
+extern "C" {
+#include "adiak_tool.h"
+}
+
+namespace axom
+{
+namespace sina
+{
+
+/**
+ * The callback function to pass to Adiak in order to write collected data to a Sina Record.
+ *
+ * To register this with Adiak, you should call adiak_register_cb, passing in
+ * a pointer to the record as the last value. Your code should look something
+ * like this:
+ *
+ * \code
+ *     axom::sina::Record record{ID{"my_id", axom::sina::IDType::Local}, "my_record_type"};
+ *     axom::sina::adiak_register_cb(1, adiak_category_all, axom::sina::adiakSinaCallback, 0, &record);
```

Comment:
That did not work for me. Instead I had to add:
#include "adiak.hpp"
extern "C" {
#include "adiak_tool.h"
#include "adiak.h"
}

And then call with:
adiak_register_cb(1, adiak_category_all, axom::sina::adiakSinaCallback, 0, &record);

Reply:
but it might be just for me.